tests/main/interfaces-iscsi-initiator/task.yaml

```
@@ -0,0 +1,95 @@
+summary: Ensure that the iscsi-initiator interface works.
+
+details: |
+  The iscsi-initiator interface allows snaps to act as iSCSI initiators,
+  enabling them to discover, connect to, and manage iSCSI targets for
+  block storage access.
+
+  A snap which defines an iscsi-initiator plug must be shown in the interfaces
+  list. The plug must not be auto-connected on install and, as usual, must be
+  able to be reconnected.
+
+  A snap declaring a plug on this interface must be able to access iSCSI
+  configuration files, ConfigFS for Linux-IO (LIO) target management,
+  and communicate with the iSCSI daemon.
+
+# Disable tests for core environments, needs to install additional packages
+systems: [-ubuntu-core-*]
+
+environment:
+  TEST_SNAP: test-iscsi-initiator-interfaces
+
+prepare: |
+  echo "Create test iSCSI configuration directory structure"
+  "$TESTSTOOLS"/fs-state mock-dir /etc/iscsi/nodes
+  tests.cleanup defer "$TESTSTOOLS"/fs-state restore-dir /etc/iscsi/nodes
+  "$TESTSTOOLS"/fs-state mock-file /etc/iscsi/initiatorname.iscsi
+  tests.cleanup defer "$TESTSTOOLS"/fs-state restore-file /etc/iscsi/initiatorname.iscsi
+  "$TESTSTOOLS"/fs-state mock-file /etc/iscsi/iscsid.conf
+  tests.cleanup defer "$TESTSTOOLS"/fs-state restore-file /etc/iscsi/iscsid.conf
+  echo "InitiatorName=iqn.1993-08.org.debian:01:abcdef123" > /etc/iscsi/initiatorname.iscsi
+  echo "node.startup = automatic" > /etc/iscsi/iscsid.conf
+
+  snap install "$TEST_SNAP" --channel=latest/edge
+
+  tests.pkgs install socat
+  systemd-run --unit socat-for-test.service --collect -- socat abstract-listen:ISCSIADM_ABSTRACT_NAMESPACE,fork - &
```

this already runs in the background:
```suggestion
  systemd-run --unit socat-for-test.service --collect -- socat abstract-listen:ISCSIADM_ABSTRACT_NAMESPACE,fork -
```